### ISSUE TITLE
reloc/build_rpm.sh problems

### ISSUE THREAD

**Issue by duarten** [scylladb/scylla#3956]
`reloc/build_rpm.sh` doesn't seem to be working:

```
~/code/scylla-dev $ reloc/build_rpm.sh --dist
'/home/duarten/code/scylla-dev/build/redhat/SOURCES/scylla-package.tar.gz' => '/home/duarten/code/scylla-dev/build/release/scylla-package.tar.gz'
error: line 3: Illegal sequence ".." in: Release:        0.20181203..el7
```

Additionally, I'm having trouble getting it to accept the `jobs` argument:

```
~/code/scylla-dev $ reloc/build_rpm.sh --jobs 4
'/home/duarten/code/scylla-dev/build/redhat/SOURCES/scylla-package.tar.gz' => '/home/duarten/code/scylla-dev/build/release/scylla-package.tar.gz'
error: Macro %_smp_mflags has empty body
```

**Comment by syuu1228** [comment]
@duarten Could you upload scylla-package.tar.gz somewhere for reploducing the issue?

**Comment by duarten** [comment]
In the meanwhile I removed it, but I just ran on master:

```
ninja-build build/release/scylla-package.tar.gz -j16
reloc/build_rpm.sh
```

On Fedora 29.

**Comment by psarna** [comment]
One more:
```
RPM build errors:
    build-id found in /home/sarna/repo/scylla/build/redhat/BUILDROOT/scylla-666.development-0.20181204.1b7ded14b.el7.x86_64/opt/scylladb/libexec/iotune.bin too small
    Generating build-id links failed
```

**Comment by duarten** [comment]
Oh, it worked for me now. I was missing git on that directory. Maybe we can check for it and add a more informative error.

**Comment by syuu1228** [comment]
I found a bug on dist/redhat/build_rpm.sh that causes "error: Macro %_smp_mflags has empty body" when we specified --jobs <n>, it can be fix, but I think we should just drop --jobs option instead of fixing the issue.
Because the option was intended for specify "-jN" option to ninja-build, but since we merged relocatable package we build nothing while generating .rpm:
https://github.com/scylladb/scylla/blob/master/dist/redhat/scylla.spec.mustache#L61

So --jobs option affects nothing, better to drop.
Now we have reloc/build_reloc.sh to generate relocatable package, it has --jobs option so we should you this instead of passing it to reloc/build_rpm.sh.

BTW I think error message should improve if there is a way to do, and also if there is argument parser issue we should be fix.
